packages/ledger/src/index.ts

```diff
@@ -0,0 +1,261 @@
+
+        let currentChain: Chain | undefined
+        const scanAccounts = async ({
+          derivationPath,
+          chainId,
+          asset
+        }: ScanAccountsOptions): Promise<Account[]> => {
+          currentChain = chains.find(({ id }) => id === chainId)
+          // TODO: Maybe throw error here
+          if (!currentChain) return []
+
+          const { publicKey, chainCode } = await eth.getAddress(
+            derivationPath,
+            false,
+            true // set to true to return chainCode
+          )
+
+          const { compress } = (await import('eth-crypto')).publicKey
+
+          return getAddresses(
+            {
+              publicKey: compress(publicKey),
+              chainCode: chainCode ?? '',
+              derivationPath: derivationPath
+            },
+            asset,
+            currentChain
+          )
+        }
+
+        const getAccounts = async () =>
+          accounts ??
+          (accounts = await accountSelect({
+            basePaths: DEFAULT_BASE_PATHS,
+            assets,
+            chains,
+            scanAccounts,
+            walletIcon: '<svg></svg>'
+          }))
+
+        const ledgerProvider = {}
+
+        const provider = createEIP1193Provider(ledgerProvider, {
+          eth_requestAccounts: async baseRequest => {
+            // Triggers the account select modal if no accounts have been selected
+            const [{ address }] = await getAccounts()
+            return [address]
+          },
+          eth_accounts: async baseRequest => {
+            return accounts?.[0]?.address ? [accounts[0].address] : []
+          },
+          eth_chainId: async baseRequest => {
+            return currentChain?.id ?? ''
+          },
+          eth_getBalance: async (baseRequest, [address, block]) => {
+            return currentChain?.rpcUrl
+              ? getBalance(address, currentChain?.rpcUrl, block)
+              : '0x'
+          },
+          eth_signTransaction: async (baseRequest, [transactionObject]) => {
+            const common = new Common({
+              chain: currentChain?.id || 1,
+              // Berlin is the minimum hardfork that will allow for EIP1559
+              hardfork: Hardfork.Berlin,
+              // List of supported EIPS
+              eips: [1559]
+            })
+            try {
+              const transaction = Transaction.fromTxData(
+                {
+                  ...transactionObject
+                },
+                { common }
+              )
+              let unsignedTx = transaction.getMessageToSign(false)
+
+              // If this is not an EIP1559 transaction then it is legacy and it needs to be
+              // rlp encoded before being passed to ledger
+              if (!transaction.supports(Capability.EIP1559FeeMarket)) {
+                unsignedTx = rlp.encode(unsignedTx)
+              }
+
+              const [{ derivationPath }] = accounts || [{ derivationPath: '' }]
```

Comment:
@aaronbarnardsound The JSON RPC API for eth_signTransaction has a `from` address. Do we just ignore this field and use the user selected account address as the `from` field? 
In current onboard it looks like we don't touch the `from` field on the transaction object and assume that the `from` field is the same as the first path. ([source](https://github.com/blocknative/onboard/blob/e3f04c43ed705c56099a91e5d2be704fb38599e8/src/modules/select/wallets/ledger.ts#L407))

Reply:
I think if there is a from field, we map that to the derivation path and use that, if not just use the first path